File: impl/go/alter_table_test.go
package jed

import (
	"bytes"
	"testing"
)

func TestAlterAddColumnRewriteMatchesFreshTableBytes(t *testing.T) {
	t.Parallel()
	altered := memDB().Session(SessionOptions{})
	uqRun(t, altered, "CREATE TABLE t (id i32 PRIMARY KEY)")
	uqRun(t, altered, "INSERT INTO t VALUES (1), (2)")
	uqRun(t, altered, "ALTER TABLE t ADD v i32 DEFAULT 7")

	fresh := memDB().Session(SessionOptions{})
	uqRun(t, fresh, "CREATE TABLE t (id i32 PRIMARY KEY, v i32 DEFAULT 7)")
	uqRun(t, fresh, "INSERT INTO t (id) VALUES (1), (2)")

	a, err := altered.ToImage(8192, 1)
	if err != nil {
		t.Fatal(err)
	}
	b, err := fresh.ToImage(8192, 1)
	if err != nil {
		t.Fatal(err)
	}
	if !bytes.Equal(a, b) {
		t.Fatal("ADD COLUMN rewrite differs from an equivalent fresh table")
	}
}

File: impl/ts/tests/alter_table.test.ts
// Byte-level ALTER TABLE rewrite checks that the shared SQL corpus cannot express.

import assert from "node:assert/strict";
import { test } from "node:test";
import { memDb } from "./mem_db.ts";
import { queryOutcome } from "./util.ts";

test("ADD COLUMN rewrite matches equivalent fresh-table bytes", () => {
  const altered = memDb().session();
  queryOutcome(altered, "CREATE TABLE t (id i32 PRIMARY KEY)");
  queryOutcome(altered, "INSERT INTO t VALUES (1), (2)");
  queryOutcome(altered, "ALTER TABLE t ADD v i32 DEFAULT 7");

  const fresh = memDb().session();
  queryOutcome(fresh, "CREATE TABLE t (id i32 PRIMARY KEY, v i32 DEFAULT 7)");
  queryOutcome(fresh, "INSERT INTO t (id) VALUES (1), (2)");

  assert.deepEqual(altered.toImage(8192, 1n), fresh.toImage(8192, 1n));
});
